vscode-extension/src/utils.ts

```diff
@@ -0,0 +1,64 @@
+import * as path from "path";
+import * as fs from "fs/promises";
+
+async function exists(file: string): Promise<boolean> {
+  return fs
+    .stat(file)
+    .then(() => true)
+    .catch(() => false);
+}
+
+function getBinaryPathRelativeToPackageJson() {
+  let binaryPathRelativeToPackageJson;
+  if (process.platform === "darwin" && process.arch === "x64") {
+    binaryPathRelativeToPackageJson = path.join("macos-x64", "relay");
+  } else if (process.platform === "darwin" && process.arch === "arm64") {
+    binaryPathRelativeToPackageJson = path.join("macos-arm64", "relay");
+  } else if (process.platform === "linux" && process.arch === "x64") {
+    binaryPathRelativeToPackageJson = path.join("linux-x64", "relay");
+  } else if (process.platform === "win32" && process.arch === "x64") {
+    binaryPathRelativeToPackageJson = path.join("win-x64", "relay.exe");
+  } else {
+    binaryPathRelativeToPackageJson = null;
+  }
+
+  if (binaryPathRelativeToPackageJson) {
+    return path.join(
+      ".",
+      "node_modules",
+      "relay-compiler",
+      binaryPathRelativeToPackageJson
+    );
+  }
+
+  return null;
+}
+
+export async function findRelayBinary(
+  rootPath: string
+): Promise<string | null> {
+  const binaryPathRelativeToPackageJson = getBinaryPathRelativeToPackageJson();
+
+  let currentPath = rootPath;
+  while (true) {
+    let possibleBinaryPath = path.join(
+      currentPath,
+      binaryPathRelativeToPackageJson
+    );
+
+    if (await exists(possibleBinaryPath)) {
+      return possibleBinaryPath;
+    }
+
+    let nextPath = path.normalize(path.join(currentPath, ".."));
+
+    // Eventually we'll get to `/` and get stuck in a loop.
+    if (nextPath === currentPath) {
```

Comment:
Is this copied from some extension? If not, it might be worthing seeing if we can find someone to test this on Windows just to double check. I assume it should work, but an infinite loop is also a bad error to have.

Reply:
yeah I'm definitely making an assumption here. i'll try to run some tests on my gaming pc tonight

Reply:
If you have any question about it, you could add a counter and increment each loop and bail if it reaches something absurd like 5,000.

Reply:
good call

Reply:
<img width="383" alt="image" src="https://user-images.githubusercontent.com/6754223/161815774-9fd2bcc1-f85b-450d-8966-404fc5c42d11.png">

Reply:
looks like it works on windows. but i'll keep the counter in as a fail safe just in case.